app/controllers/api/v0/api_controller.rb

```diff
@@ -43,15 +43,24 @@ def error_not_found
     render json: { error: "not found", status: 404 }, status: :not_found
   end
 
-  def authenticate_with_api_key_or_current_user
+  def authenticate!
+    if doorkeeper_token
```

Comment:
great!

we check first for the oauth2 token, then for the api key, then for the cookie.

Reply:
Should I be adding in a `else if current_user`? 

`#authenticate!` only checks for access token and api key.

Reply:
What about?

```ruby
def authenticate!
  if doorkeeper_token
    @user = User.find(doorkeeper_token.resource_owner_id)
    return error_unauthorized unless @user
  elsif request.headers["api-key"]
    authenticate_with_api_key!
  elsif current_user
    @user = current_user
  else
    error_unauthorized
  end
end
```

Reply:
updated.